### PR TITLE
Symbolicate Dart stacktrace on Flutter Android and iOS without debug images from native sdks

### PR DESCRIPTION
**Pull request by buenaflor**
## :scroll: Description

Dart StackTraces can be symbolicated with `Sentry.init` on Flutter iOS, macOS and Android

Uses the stacktrace header, specifically `isolateDsoBase / vmDsoBase` & `build_id` to construct a synthetic debug image which allows symbolication to work also on dart stacktraces without debug images that we get from the cocoa and android sdks.

Screenshot 2024-08-28 at 18 09 28

## :bulb: Motivation and Context

There's a need for using only sentry-dart on Flutter due to app size limitations, this should allow for some symbolication.

## :green_heart: How did you test it?
Unit test, local testing

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps
Get the code_file path somehow, is probably not possible with a pure dart setup

### CODE REVIEW

File: dart/lib/src/debug_image_extractor.dart [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    final reversedByteArray = Uint8List.fromList([
+      ...byteArray.sublist(0, 4).reversed,
+      ...byteArray.sublist(4, 6).reversed,
+      ...byteArray.sublist(6, 8).reversed,
+      ...byteArray.sublist(8, 10),
+      ...byteArray.sublist(10),
+    ]);
+
+    return Uuid.unparse(reversedByteArray);
+  }
```
Comment (buenaflor) [inline dart/lib/src/debug_image_extractor.dart]: not sure how performant this is

File: flutter/lib/src/integrations/load_native_debug_images_integration.dart [hunk 2]
```diff
@@ -4,11 +4,16 @@ import 'package:sentry/sentry.dart';
 import '../native/sentry_native_binding.dart';
 import '../sentry_flutter_options.dart';
 
+// ignore: implementation_imports
+import 'package:sentry/src/load_dart_debug_images_integration.dart'
+    show NeedsSymbolication;
+
 /// Loads the native debug image list for stack trace symbolication.
-class LoadImageListIntegration extends Integration<SentryFlutterOptions> {
+class LoadNativeDebugImagesIntegration
```
Comment (stefanosiano) [inline flutter/lib/src/integrations/load_native_debug_images_integration.dart]: Is there a valid reason to change the name of the integration?
I don't want to break users that e.g. remove the integration based on the class name

**Review by stefanosiano — COMMENTED** [review]
looks good overall!
just a question on an integration name

File: dart/lib/src/debug_image_extractor.dart (hunk 1 above)
Comment (stefanosiano) [reply dart/lib/src/debug_image_extractor.dart]: hopefully we call it only once, so it shouldn't be a problem

File: flutter/lib/src/integrations/load_native_debug_images_integration.dart (hunk 2 above)
Comment (buenaflor) [reply flutter/lib/src/integrations/load_native_debug_images_integration.dart]: now that we're introducing another debug image loading intrgration "load images list" gets kinda ambiguous 

but true probably not a good idea to change it now

I'd personally still prefer to mark it as renaming todo and change it in the next major for example

File: dart/lib/src/debug_image_extractor.dart (hunk 1 above)
Comment (buenaflor) [reply dart/lib/src/debug_image_extractor.dart]: yup we do

**Review by stefanosiano — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,11 @@
 
 ### Features
 
+ - Add `enableDartSymbolication` option to Sentry.init() for **Flutter iOS, macOS and Android** ([#2256](https://github.com/getsentry/sentry-dart/pull/2256))
+    - This flag enables symbolication of Dart stack traces when native debug images are not available.
+    - Useful when using Sentry.init() instead of SentryFlutter.init() in Flutter projects for example due to size limitations.
+    - `true` by default but automatically set to `false` when using SentryFlutter.init() because the SentryFlutter fetches debug images from the native SDK integrations.
+- Support allowUrls and denyUrls for Flutter Web ([#2227](https://github.com/getsentry/sentry-dart/pull/2227))
```
Comment (vaind) [inline CHANGELOG.md:11]: This line for another PR got duplicated here

Comment (vaind) [reply CHANGELOG.md:11]: FYI: I'll fix it in one of the other PRs while resolving merge conflicts

Comment (buenaflor) [reply CHANGELOG.md:11]: thx

**Pull request merged**